main/webapp/modules/core/scripts/project/history-panel.js

```diff
@@ -254,6 +255,23 @@ HistoryPanel.prototype._showExtractOperationsDialog = function(json) {
     frame.find('input[type="checkbox"]').prop('checked', false);
     updateJson();
   });
+  elmts.saveJsonAsFileButton.click(function() {
+    var historyJson = elmts.textarea[0].value;
+
+    downloadFile('history.json', historyJson);
+  });
+
+  var downloadFile = function(filename, content) {
```

Comment:
It could be worth adding a comment to attribute the author (StackOverflow content is CC-BY-SA).

Reply:
Absolutely!

Reply:
Done!

I have a question about commits. What is a better way to write commits? Can I use the format: closes #1234 - (short description of the commit) for each commit I produce related to this issue? Also, if we need to update the commit messages, shall I use "git push -f"?
Thanks! :)

Reply:
It is a good practice to include "closes #4498" in your commit message, but preferably after some informative description of the commit. You can also put "closes #4498" further down the commit message (not necessarily on the first line).
For instance:
```
Add download button for JSON history.

Closes #4498.
```

You can use `git push -f` on your PR branches but it is not necessary, we squash commits when merging pull requests. (Personally I even like it when people just do more commits and not force-push, that helps me review the changes since my last review)

Reply:
Thanks so much for your detailed explanation! I will keep this noted! :)